config.cc

```diff
@@ -2206,6 +2225,14 @@ QString getIndexDir() THROW_SPEC( exError )
 {
   QDir result = getHomeDir();
 
+#ifdef HAVE_X11
+  // check for existing index in old location – if it is not present use new standard location for cache
+  if ( !result.exists( "index" ) )
+  {
+    result.setPath( getCacheDir() );
+  }
+#endif
```

Comment:
I think that it would be better to limit this to `XDG_BASE_DIRECTORY_COMPLIANCE` as well. Otherwise, switching from Qt4 to Qt5 in the future would unexpectedly recreate the index files and leave the old index files wasting disk space. `XDG_BASE_DIRECTORY_COMPLIANCE` can be defined where it is first used - in the second commit. Don't forget to mention Qt5-only in the 3rd commit message.